runtime/src/gpu/rocm/gpu-rocm.c

```diff
@@ -429,6 +429,8 @@ void chpl_gpu_impl_comm_wait(void *stream) {
 void* chpl_gpu_mem_array_alloc(size_t size, chpl_mem_descInt_t description,
                                int32_t lineno, int32_t filename) {
 
+  if(!chpl_gpu_device_alloc) { return chpl_malloc(size); }
```

Comment:
FWIW, I have a branch where I simplify context management (I think I was overthinking when I wrote the current one). I wonder if that helps get rid of this. Could you add a comment here saying that this is a workaround for the context management issue you were facing.

This is OK, but it still looks like a workaround for these functions to redirect to `chpl_malloc`. At the very least, that difference between gpu/non-gpu allocations should be made earlier in the call chain.

Reply:
Sure, I've added this comment:

```
  // This line is a workaround since currently it looks like
  // `chpl_gpu_ensure_context()` fails in our ROCM runtime when there are
  // multiple devices. This function (`chpl_gpu_mem_array_alloc`) is called
  // when initializing the Chapel module for the GPU locale module
  // (chpl_gpu_device_alloc == true during this time), in this case the intent
  // is just to return host memory anyway so there's no need to switch
  // contexts.
  ```

> At the very least, that difference between gpu/non-gpu allocations should be made earlier in the call chain.

I think typically typically we do, just not for that one weird allocation we have in the gpu locale model where we end up having the current sublocale id as a GPU but still want to allocate on the host.